clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordBatch.java

```diff
@@ -67,10 +68,13 @@ public FutureRecordMetadata tryAppend(long timestamp, byte[] key, byte[] value,
         if (!this.records.hasRoomFor(key, value)) {
             return null;
         } else {
-            this.records.append(offsetCounter++, timestamp, key, value);
+            long checksum = Record.computeChecksum(timestamp, key, value, CompressionType.NONE, 0, -1);
```

Comment:
I see, makes sense, thanks. It may be interesting to see if we actually need so many `Compressor.putRecord` overloads as they make the code harder to understand.